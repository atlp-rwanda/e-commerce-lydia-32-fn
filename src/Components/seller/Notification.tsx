import React from "react";

const Notification: React.FC<{
  icon: string;
  message: string;
  time: string;
  status:boolean
}> = ({ icon, message, time,status }) => {
  return (
    <div className="flex items-center justify-between bg-white p-4 mb-4 rounded-lg shadow-md">
      <div className="flex items-center flex-grow">
        <span className="text-2xl mr-4 flex-shrink-0">{icon}</span>
        <div className="flex-grow">
          <p className= {!status?"font-bold":"font-light"}>{message}</p>
          <span className="text-sm text-gray-500">{time}</span>
        </div>
        
      </div>
    </div>
  );
};

export default Notification;
