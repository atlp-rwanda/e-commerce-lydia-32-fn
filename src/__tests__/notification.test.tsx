// Notification.test.tsx
import React from 'react';
import { render, screen } from '@testing-library/react';
import Notification from '../Components/seller/Notification';
import '@testing-library/jest-dom'; // For the matchers

describe('Notification', () => {
  const icon = 'ðŸ””';
  const message = 'New message received';
  const time = '2 minutes ago';

  it('renders the notification with bold message when status is false', () => {
    render(<Notification icon={icon} message={message} time={time} status={false} />);
    
    expect(screen.getByText(icon)).toBeInTheDocument();
    expect(screen.getByText(message)).toHaveClass('font-bold');
    expect(screen.getByText(time)).toBeInTheDocument();
  });

  it('renders the notification with light message when status is true', () => {
    render(<Notification icon={icon} message={message} time={time} status={true} />);
    
    expect(screen.getByText(icon)).toBeInTheDocument();
    expect(screen.getByText(message)).toHaveClass('font-light');
    expect(screen.getByText(time)).toBeInTheDocument();
  });

  it('renders the notification with the correct icon', () => {
    render(<Notification icon={icon} message={message} time={time} status={false} />);
    
    expect(screen.getByText(icon)).toBeInTheDocument();
  });

  it('renders the notification with the correct time', () => {
    render(<Notification icon={icon} message={message} time={time} status={false} />);
    
    expect(screen.getByText(time)).toBeInTheDocument();
  });

  it('does not apply the bold class when status is true', () => {
    render(<Notification icon={icon} message={message} time={time} status={true} />);
    
    expect(screen.getByText(message)).not.toHaveClass('font-bold');
  });

  it('does not apply the light class when status is false', () => {
    render(<Notification icon={icon} message={message} time={time} status={false} />);
    
    expect(screen.getByText(message)).not.toHaveClass('font-light');
  });

  it('displays an icon correctly', () => {
    render(<Notification icon={icon} message={message} time={time} status={false} />);
    
    expect(screen.getByText(icon)).toBeInTheDocument();
  });

  it('displays the correct message text', () => {
    render(<Notification icon={icon} message={message} time={time} status={false} />);
    
    expect(screen.getByText(message)).toBeInTheDocument();
  });

  it('displays the correct time text', () => {
    render(<Notification icon={icon} message={message} time={time} status={false} />);
    
    expect(screen.getByText(time)).toBeInTheDocument();
  });

  it('renders the notification with a different icon', () => {
    const newIcon = 'ðŸ“§';
    render(<Notification icon={newIcon} message={message} time={time} status={false} />);
    
    expect(screen.getByText(newIcon)).toBeInTheDocument();
  });

  it('renders the notification with a different message', () => {
    const newMessage = 'You have a new email';
    render(<Notification icon={icon} message={newMessage} time={time} status={false} />);
    
    expect(screen.getByText(newMessage)).toBeInTheDocument();
  });

  it('renders the notification with a different time', () => {
    const newTime = '5 minutes ago';
    render(<Notification icon={icon} message={message} time={newTime} status={false} />);
    
    expect(screen.getByText(newTime)).toBeInTheDocument();
  });

  it('applies the bold class correctly when status is false', () => {
    render(<Notification icon={icon} message={message} time={time} status={false} />);
    
    expect(screen.getByText(message)).toHaveClass('font-bold');
  });

  it('applies the light class correctly when status is true', () => {
    render(<Notification icon={icon} message={message} time={time} status={true} />);
    
    expect(screen.getByText(message)).toHaveClass('font-light');
  });

  it('renders multiple notifications correctly', () => {
    const notifications = [
      { icon: 'ðŸ””', message: 'First message', time: '1 minute ago', status: false },
      { icon: 'ðŸ“§', message: 'Second message', time: '2 minutes ago', status: true },
    ];

    notifications.forEach((notif) => {
      render(<Notification icon={notif.icon} message={notif.message} time={notif.time} status={notif.status} />);
      
      expect(screen.getByText(notif.icon)).toBeInTheDocument();
      expect(screen.getByText(notif.message)).toBeInTheDocument();
      expect(screen.getByText(notif.time)).toBeInTheDocument();
    });
  });
});
