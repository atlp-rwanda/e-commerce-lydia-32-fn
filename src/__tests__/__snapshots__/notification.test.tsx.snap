// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Notification Component > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex items-center justify-between bg-white p-4 mb-4 rounded-lg shadow-md"
  >
    <div
      class="flex items-center flex-grow"
    >
      <span
        class="text-2xl mr-4 flex-shrink-0"
      >
        ðŸ””
      </span>
      <div
        class="flex-grow"
      >
        <p
          class="font-bold"
        >
          You have a new message
        </p>
        <span
          class="text-sm text-gray-500"
        >
          2 hours ago
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
